apps/email/js/cards/message_list.js

```diff
@@ -1753,11 +1757,15 @@ return [
 
         this.editBtn.disabled = true;
         this.refreshBtn.dataset.state = 'synchronizing';
+        this.refreshBtn.setAttribute('role', 'progressbar');
+        mozL10n.setAttributes(this.refreshBtn, 'refresh-progress');
       } else {
         // After sync, the edit button should remain disabled only if
         // the list is empty.
         this.editBtn.disabled = this.isEmpty();
         this.refreshBtn.dataset.state = 'synchronized';
+        this.refreshBtn.removeAttribute('role');
+        mozL10n.setAttributes(this.refreshBtn, 'refresh-button');
```

Comment:
Since the refreshBtn update now needs three steps for each state, and this needs to be done at different places, how about a `setRefreshState(state)` that does the dataset, setAttribute work for each cases then call that function from these points?